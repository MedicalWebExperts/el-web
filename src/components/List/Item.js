import React from 'react';
import { string, shape } from 'prop-types';
import theme from '../../styles';

const propTypes = {
  text: string.isRequired,
  styles: shape({}),
};

const defaultProps = {
  styles: {},
};

const Item = ({ text, styles }) => (
  <li style={styles}>
    <style jsx>
      {`
        li {
          padding: 0 0 0 20px;
          position: relative;
          color: ${theme.colors.default};
          font-family: ${theme.font.primary};
          font-size: ${theme.font.size}px;
          line-height: 1.6;
        }
        li:before {
          content: '';
          position: absolute;
          left: 0;
          top: 7px;
          right: auto;
          width: 8px;
          height: 8px;
          border-bottom: solid 1px currentColor;
          border-left: solid 1px currentColor;
          transform: rotate(-135deg);
          color: #4a90e2;
        }
      `}
    </style>
    {text}
  </li>
);

Item.propTypes = propTypes;
Item.defaultProps = defaultProps;
export default Item;
