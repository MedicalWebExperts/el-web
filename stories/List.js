import React from 'react';
import { storiesOf } from '@storybook/react';

import { List, ListItem } from '..';
import { providers } from '../mocks/providers';
import { locations } from '../mocks/locations';

export default () => {
  storiesOf('List', module).add('List Providers', () => (
    <List
      items={providers.map(item => (
        <ListItem
          type={item.type}
          avatar={item.avatar}
          name={item.name}
          title={item.title}
          specialty={item.specialty}
          location={item.location}
          description={item.description}
          secondaryDescription={item.secondaryDescription}
          action={item.action}
          actionText={item.actionText}
        />
      ))}
    />
  ));
  storiesOf('List', module).add('List Locations', () => (
    <List
      items={locations.map(item => (
        <ListItem
          type={item.type}
          avatar={item.avatar}
          name={item.name}
          address={item.address}
          city={item.city}
          state={item.state}
          action={item.action}
          actionText={item.actionText}
        />
      ))}
    />
  ));
};
