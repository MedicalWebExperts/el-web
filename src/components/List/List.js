import React from 'react';
import { arrayOf, shape, string } from 'prop-types';
import Item from './Item';

const propTypes = {
  data: arrayOf(string).isRequired,
  styles: shape({}),
};

const defaultProps = {
  styles: {},
};

const List = ({ data, styles }) => (
  <ul style={styles}>
    <style jsx>
      {`
        ul {
          list-style: none;
          padding: 0;
          margin: 0;
        }
      `}
    </style>

    {data.map((item, index) => (
      <Item text={item} key={index} />
    ))}
  </ul>
);

List.propTypes = propTypes;
List.defaultProps = defaultProps;
export default List;
