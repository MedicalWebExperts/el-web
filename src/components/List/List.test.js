import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';

const defaultProps = {
  data: [
    'American Medical Association',
    'Texas Medical Association',
    'American Academy of Family Physicians',
  ],
};

const customStyles = {
  styles: { margin: 20 },
};

describe('Component List Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<List {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer
      .create(<List {...Object.assign({}, defaultProps, customStyles)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
