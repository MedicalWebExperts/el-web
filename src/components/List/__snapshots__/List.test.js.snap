// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component List Snapshot using custom styles 1`] = `
<ul
  className="jsx-1253930413"
  style={
    Object {
      "margin": 20,
    }
  }
>
  <li
    className="jsx-1154723179"
    style={Object {}}
  >
    American Medical Association
  </li>
  <li
    className="jsx-1154723179"
    style={Object {}}
  >
    Texas Medical Association
  </li>
  <li
    className="jsx-1154723179"
    style={Object {}}
  >
    American Academy of Family Physicians
  </li>
</ul>
`;

exports[`Component List Snapshot without arguments, take the default values 1`] = `
<ul
  className="jsx-1253930413"
  style={Object {}}
>
  <li
    className="jsx-1154723179"
    style={Object {}}
  >
    American Medical Association
  </li>
  <li
    className="jsx-1154723179"
    style={Object {}}
  >
    Texas Medical Association
  </li>
  <li
    className="jsx-1154723179"
    style={Object {}}
  >
    American Academy of Family Physicians
  </li>
</ul>
`;
